tests/pytests/l3_dump_test.py

```diff
@@ -0,0 +1,545 @@
+    binary = L3RootDir + '/build/' + BUILD_MODE + '/bin/unit/l3_dump.py-test'
+    exec_rv = exec_binary(binary)
+    assert exec_rv is True
+
+    # Invoke the L3-dump script to unpack the slow-log-entries.
+    (nentries, tid_list, loc_list, msg_list, arg1_list, arg2_list) \
+        = l3_dump.do_main(['/tmp/l3.c-small-unit-test.dat', binary],
+                          return_logentry_lists = True)
+
+    # -----------------------------------------------------------------------
+    # Here is where it gets interesting. Hard-coded verification of expected
+    # message and arguments, as unpacked by utility. This is one example of
+    # verification using hard-coded expected-outputs lists.
+    # -----------------------------------------------------------------------
+    exp_msg_list = [  'Simple-log-msg-Args(1,2)'
+                    , 'Simple-log-msg-Args(3,4)'
+                    , 'Potential memory overwrite (addr, size)'
+                    , 'Invalid buffer handle (addr)'
+                   ]
+    assert msg_list == exp_msg_list
+
+    exp_arg1_list = [ 1, 3, int('0xdeadbabe', 16), int('0xbeefabcd', 16) ]
+    assert arg1_list == exp_arg1_list
+
+    exp_arg2_list = [ 2, 4, 1024, 0 ]
+    assert arg2_list == exp_arg2_list
+
+    verify_rv = verify_output_lists(nentries, len(exp_msg_list),
+                                    tid_list, loc_list,
+                                    msg_list,
+                                    arg1_list, arg2_list)
+    assert verify_rv is True
+
+    # Unit-tests are currently not enabled to run with LOC_ENABLED env-var
+    assert verify_loc_field_is_empty(loc_list) is True
+
+    # Invoke the L3-dump script to unpack the fast-log-entries.
+    (nentries, tid_list, loc_list, msg_list, arg1_list, arg2_list) \
+        = l3_dump.do_main(['/tmp/l3.c-fast-unit-test.dat', binary],
+                          return_logentry_lists = True)
+
+    exp_msg_list = [  'Fast-log-msg: Args(1,2)'
+                    , 'Fast-log-msg: Args(3,4)'
+                    , 'Fast-log-msg: Args(10,20)'
+                    , 'Fast-log-msg: Potential memory overwrite (addr, size)'
+                    , 'Fast-log-msg: Invalid buffer handle (addr)'
+                   ]
+    assert msg_list == exp_msg_list
+
+    exp_arg1_list = [ 1, 3, 10, int('0xdeadbabe', 16), int('0xbeefabcd', 16) ]
+    assert arg1_list == exp_arg1_list
+
+    exp_arg2_list = [ 2, 4, 20, 1024, 0 ]
+    assert arg2_list == exp_arg2_list
+
+    verify_rv = verify_output_lists(nentries, len(exp_msg_list),
+                                    tid_list, loc_list,
+                                    msg_list,
+                                    arg1_list, arg2_list)
+    assert verify_rv is True
+
+    # Unit-tests are currently not enabled to run with LOC_ENABLED env-var
+    assert verify_loc_field_is_empty(loc_list) is True
+
+# #############################################################################
+def test_c_test_dump_log_entries():
```

Comment:
The structure of most test cases here onwards is similar.

- We do some initial boilerplate stuff.
- Build the program
- Execute the resulting binary, generating the L3-log-dump file
- Call the verification method which will do the required cross-checking, for default case (w/ no LOC encoding) and with default LOC-encoding

Reply:
Did you consider using pexpect here? Can be useful for this sort of testing.